server/applicationset/applicationset.go

```diff
@@ -366,6 +365,42 @@ func (s *Server) ResourceTree(ctx context.Context, q *applicationset.Application
 	return s.buildApplicationSetTree(a)
 }
 
+func (s *Server) Generate(ctx context.Context, q *applicationset.ApplicationSetGenerateRequest) (*applicationset.ApplicationSetGenerateResponse, error) {
+	appset := q.GetApplicationSet()
+
+	if appset == nil {
+		return nil, fmt.Errorf("error creating ApplicationSets: ApplicationSets is nil in request")
+	}
+	namespace := s.appsetNamespaceOrDefault(appset.Namespace)
+
+	if !s.isNamespaceEnabled(namespace) {
+		return nil, security.NamespaceNotPermittedError(namespace)
+	}
+	projectName, err := s.validateAppSet(appset)
+	if err != nil {
+		return nil, fmt.Errorf("error validating ApplicationSets: %w", err)
+	}
+	if err := s.checkCreatePermissions(ctx, appset, projectName); err != nil {
+		return nil, fmt.Errorf("error checking create permissions for ApplicationSets %s : %w", appset.Name, err)
+	}
+
+	logs := bytes.NewBuffer(nil)
+	logger := log.New()
+	logger.SetOutput(logs)
+
+	res := &applicationset.ApplicationSetGenerateResponse{}
+
+	apps, err := s.generateApplicationSetApps(ctx, logger.WithField("applicationset", appset.Name), *appset, namespace)
+	if err != nil {
+		res.ErrorLog = fmt.Sprintf("unable to generate Applications of ApplicationSet: %v\n", err)
```

Comment:
why not return error and build error message on cli , base on this error ? In any case you will need process error on client part, to show reasonable error when request or permissions are incorrect, i think it will be more consistent to do it also when you cant generate applications

Reply:
Makes sense. `ErrorLog` just duplicates error - removed it.